src/Azure.Functions.ArtifactAssembler/ArtifactAssembler.cs

```diff
@@ -161,12 +198,150 @@ private async Task CreateVisualStudioCoreToolsAsync()
+            Directory.CreateDirectory(cliCoreToolsTargetArtifactDir);
+            string outOfProcVersion = string.Empty;
+            string inProcVersion = string.Empty;
+            string outOfProcArtifactDirPath = string.Empty;
+            string inProc8ArtifactDirPath = string.Empty;
+
+            var packTasks = _cliArtifacts.Select(async artifactName =>
+            {
+                // If we are running this for the first time, extract the directory path and out of proc version
+                if (String.IsNullOrEmpty(outOfProcArtifactDirPath))
+                {
+                    var outOfProcResponse = GetArtifactDirectoryAndVersionNumber(_outOfProcExtractedRootDir, artifactName);
+                    outOfProcArtifactDirPath = outOfProcResponse[0];
+                    outOfProcVersion = outOfProcResponse[1];
+                }
+                else
+                {
+                    var artifactNameWithVersion = $"{artifactName}.{outOfProcVersion}";
+                    outOfProcArtifactDirPath = Path.Combine(_outOfProcExtractedRootDir, artifactNameWithVersion);
+                }
+
+                // Create a new directory to store the oop core tools with in-proc8 and in-proc6 files.
+                var outOfProcArtifactName = Path.GetFileName(outOfProcArtifactDirPath);
+                var consolidatedArtifactDirPath = Path.Combine(cliCoreToolsTargetArtifactDir, outOfProcArtifactName);
+                Directory.CreateDirectory(consolidatedArtifactDirPath);
+
+                // Copy oop host files
+                var coreToolsHostArtifactDirPath = Path.Combine(_outOfProcExtractedRootDir, outOfProcArtifactName);
+                EnsureArtifactDirectoryExist(coreToolsHostArtifactDirPath);
+                var outOfProcCopyTask = Task.Run(() => FileUtilities.CopyDirectory(coreToolsHostArtifactDirPath, consolidatedArtifactDirPath));
+
+                // If we are running this for the first time, extract the directory path and out of proc version
+                if (String.IsNullOrEmpty(inProc8ArtifactDirPath))
+                {
+                    // Get the version number from the in-proc build since it will be different than out-of-proc
+                    var inProc8Response = GetArtifactDirectoryAndVersionNumber(_inProc8ExtractedRootDir, artifactName);
+                    inProc8ArtifactDirPath = inProc8Response[0];
+                    inProcVersion = inProc8Response[1];
+                }
+                else
+                {
+                    var artifactNameWithVersion = $"{artifactName}.{inProcVersion}";
+                    inProc8ArtifactDirPath = Path.Combine(_inProc8ExtractedRootDir, artifactNameWithVersion);
+                }
+
+                // Rename inproc8 directory to have the same version as the out-of-proc artifact before copying
+                string newInProc8ArtifactDirPath = RenameInProcDirectory(inProc8ArtifactDirPath, outOfProcVersion);
```

Comment:
Why do we need this rename operation ? Aren't we eventually copying it to a directory with static name (`InProc8DirectoryName` in line 260) ?

Reply:
we need this method so that we can extract the version number from the inproc8 version (since we need the full name/version to copy the files over from one directory to another)